Stack.rb

```diff
@@ -0,0 +1,72 @@
+class Stack
+  def initialize
+    @head = nil
+    @head = @tail
+  end
+
+  def is_empty?
+  	if @head == nil
+  		true
+  	else
+  		false
+  	end
+  end
+
+  def peek
+  	if @head == nil
+  		puts "Stack is empty."
```

Comment:
I think you might want to return here since your return on line 23 is outside the if else block